src/main/java/com/google/cloud/discotoproto3converter/proto3/Field.java

```diff
@@ -87,16 +89,20 @@ public boolean equals(Object o) {
     if (o == null || getClass() != o.getClass()) {
       return false;
     }
+    if (!super.equals(o)) {
```

Comment:
Aren't the two checks before this one are already done by `super.equals()`? (I'm not sure how `getClass` works with polymorphism, but I think it does the right thing). In that case, you can get rid of them and rely on the `super` check here. Same thing in `Message.java`

Reply:
This is the standard pattern for implementing `equals()` method in java. in fact, this particular method was generated. You are right that the checks are made by super, but it is not guaranteed to be like that and may change potentially.  I.e. the child class not supposed to know the implementation details of the super class.

Reply:
Sure, but if the contract for `equals()` is clear, then it can rely on that...

OK, no biggie.